pytools/__init__.py

```diff
@@ -2997,6 +3003,49 @@ def unique(seq: Iterable[T]) -> Iterator[T]:
     :func:`itertools.groupby` (which removes consecutive duplicates)."""
     return iter(dict.fromkeys(seq))
 
+
+def unique_difference(*args: Iterable[T]) -> Iterator[T]:
```

Comment:
Would it make sense to have these return `Collection[T]` (and skip the `iter` call)?

Reply:
Hmm, I'm not sure what to return instead of the `iter` - a list or tuple? We can't just return the dictionary I guess.

Reply:
9ece452 is a version of these functions returning lists. I don't know if that's better.

Reply:
No, I thought you would just return the dictionary. A dictionary with key type `T` should count as a collection of type `T` I think.

Reply:
I.e. no need to build a separate list.

Reply:
What do you think of [7b352f3](https://github.com/inducer/pytools/pull/257/commits/7b352f3103f55b627cbf550d404b408638f05354)?

Reply:
LGTM!